packages/jazz-tools/src/auth/AuthSecretStorage.ts

```diff
@@ -0,0 +1,109 @@
+import { AgentSecret } from "cojson";
+import type { Account } from "../coValues/account.js";
+import type { ID } from "../internal.js";
+import { AuthCredentials } from "../types.js";
+import KvStoreContext from "./KvStoreContext.js";
+
+const STORAGE_KEY = "jazz-logged-in-secret";
+
+export type AuthSetPayload = {
+  accountID: ID<Account>;
+  secretSeed?: Uint8Array;
+  accountSecret: AgentSecret;
+  provider: "anonymous" | "clerk" | "demo" | "passkey" | "passphrase" | string;
+};
+
+export class AuthSecretStorage {
+  private listeners: Set<(isAuthenticated: boolean) => void>;
+  public isAuthenticated: boolean;
+
+  constructor() {
+    this.listeners = new Set();
+    this.isAuthenticated = false;
+  }
```

Comment:
Made this a class to handle the `isAuthenticated` state here and not make it global.

The change was driven by the switch to the async API that made reading from storage not possible in the init phase of the UI components.